addons/point_of_sale/models/pos_session.py

```diff
@@ -313,6 +302,522 @@ def action_pos_session_close(self):
+            # create statement line for each payment method
+            payments = statement_payments.filtered(lambda p: p.payment_method_id == payment_method)
+            currency = payment_method.cash_journal_id.currency_id or company.currency_id
+            amount_converter = lambda payment: session_currency._convert(payment.amount, currency, company, payment.payment_date)
+            return {
+                'date': fields.Date.context_today(self),
+                'amount': sum(payments.mapped(amount_converter)) if payments else 0.0,
+                'name': session.name,
+                'statement_id': statement.id,
+                'account_id': payment_method.receivable_account_id.id,
+            }
+
+        cash_payment_methods = [pm for pm in session.payment_method_ids if pm.is_cash_count and (pm.cash_journal_id == statement.journal_id)]
+        statement_lines_args = [create_vals(pm) for pm in cash_payment_methods]
+        return self.env['account.bank.statement.line'].create([args for args in statement_lines_args if args.get('amount')])
+
+    @api.model
+    def _get_invoiced_receivable_lines_vals(self, account_move, invoiced_orders, is_using_company_currency):
+        if len(invoiced_orders) == 0:
+            return []
+
+        company = invoiced_orders[0].session_id.company_id
+        session_currency = invoiced_orders[0].session_id.currency_id
+        company_currency = company.currency_id
+
+        def get_order_receivable_account_id(order):
+            account_def = self.env['ir.property'].get('property_account_receivable_id', 'res.partner')
+            order_account = order.partner_id.property_account_receivable_id or account_def
+            return order_account.id
+
+        def generate_move_line_vals(account_id, amount, amount_converted):
+            partial_args = {'account_id': account_id, 'move_id': account_move.id, 'name': 'From invoiced orders'}
+            return self._credit_amounts(partial_args, amount, amount_converted, session_currency, is_using_company_currency)
+
+        def compute_total_amounts(orders):
+            # Use `amount_total` (w/c includes tax) field since this will be
+            # reconciled with receivable item in an invoice move which counts taxes.
+            amount_converter = lambda order: session_currency._convert(order.amount_total, company_currency, company, order.date_order)
+            total_amount = sum(order.amount_total for order in orders)
+            total_amount_converted = (not is_using_company_currency) and sum(amount_converter(order) for order in orders)
+            return total_amount, total_amount_converted
+
+        grouped_invoiced_orders = groupby(invoiced_orders, key=get_order_receivable_account_id)
+        account_amounts = [(account_id, compute_total_amounts(orders)) for account_id, orders in grouped_invoiced_orders]
+        return [generate_move_line_vals(account_id, *amounts) for account_id, amounts in account_amounts]
```

Comment:
maybe it is more clear with a simple loop that will append a list with the values returned by 'self._credit_amounts(partial_args, amount, amount_converted, session_currency, is_using_company_currency)'